### PR TITLE
[fix][fn] Support multiple input topics for Go runtime

### LINKED ISSUE #19981: [Bug] Go Pulsar functions recieves message from only one topic even when provided multiple input  topics
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

OS: Ubuntu 20.04(LTS)
Pulsar: 2.10.1

### Minimal reproduce step

Create a pulsar function in golang:

```
package main

import (
	"context"
	"fmt"

	"github.com/apache/pulsar/pulsar-function-go/pf"
)

func HandleResponse(ctx context.Context, in []byte) error {
	fmt.Printf("Processed: %s", string(in))
	return nil
}

func main() {
	pf.Start(HandleResponse)
}
```

Build and pass the file to pulsar-admin:

```
./pulsar-admin functions localrun \
     --go /pulsar/bin/pulsarFuncGolang \
     --inputs persistent://public/default/sourcetopic1,persistent://public/default/sourcetopic2 \
     --tenant public \
     --namespace default \
     --name flogoPulsarFunction
```

Publish message on both the topics, we could only see the function logs of recieved messages on the last topic mentioned (which is sourcetopic2 in the example above)

### What did you expect to see?

Pulsar function should listen to all the topics mentioned

### What did you see instead?

Pulsar function listens to only the last topic mentioned

### Anything else?

_No response_

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiangpengcheng**
Fixes #19981

Master Issue: #xyz

PIP: #xyz 

### Motivation

The Go runtime doesn't support multi-input topics, 

### Modifications

Add a map<string, string> field `SourceInputSpecs` to the `GoInstanceConfig`, while the key is topic name, and value is the json string of `ConsumerSpec`, so that the Go instance can unmarshal it to a `map[string]ConsumerSpec` and support multiple input topics

### Verifying this change

- [x] Make sure that the change passes the CI checks.

- [x] This change added tests and can be verified as follows:

  - *Added integration tests for GO with multiple input topics*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/jiangpengcheng/pulsar/pull/10

### CODE REVIEW

**Review by freeznet — APPROVED** [review]

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java [hunk 1]
```diff
@@ -682,23 +689,12 @@ protected void testExclamationFunction(Runtime runtime,
...
-                    .subscriptionName("test-sub")
-                    .subscribe();
-            inputTopicName = inputTopicName + ".*";
+        if (isTopicPattern || multipleInput) {
+            if (isTopicPattern) {
+                inputTopicName = inputTopicName + ".*";
+            } else {
+                inputTopicName = inputTopicName + "1," + inputTopicName + "2";
+            }
         }
```
Comment (cbornet) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java:698]:
```suggestion
        if (isTopicPattern) {
            inputTopicName = inputTopicName + ".*";
        } else if (multipleInput) {
            inputTopicName = inputTopicName + "1," + inputTopicName + "2";
        }
    }
```

**Review by cbornet — CHANGES_REQUESTED** [review]

File: pulsar-function-go/conf/conf.go [hunk 2]
```diff
@@ -50,15 +50,17 @@ type Conf struct {
...
 	SubscriptionType     int32  `json:"subscriptionType" yaml:"subscriptionType"`
 	TimeoutMs            uint64 `json:"timeoutMs" yaml:"timeoutMs"`
 	SubscriptionName     string `json:"subscriptionName" yaml:"subscriptionName"`
 	CleanupSubscription  bool   `json:"cleanupSubscription"  yaml:"cleanupSubscription"`
 	SubscriptionPosition int32  `json:"subscriptionPosition" yaml:"subscriptionPosition"`
 	//source input specs
+	SourceInputSpecs map[string]string `json:"sourceInputSpecs" yaml:"sourceInputSpecs"`
+	// for backward compatibility
 	SourceSpecTopic            string `json:"sourceSpecsTopic" yaml:"sourceSpecsTopic"`
 	SourceSchemaType           string `json:"sourceSchemaType" yaml:"sourceSchemaType"`
```
Comment (nlu90) [inline pulsar-function-go/conf/conf.go:65]: you should mark `Deprecated` for fields that's not intended for use

**Review by nlu90 — APPROVED** [review]

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java (hunk 1 above)
Comment (jiangpengcheng) [reply tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java]: updated

File: pulsar-function-go/conf/conf.go (hunk 2 above)
Comment (jiangpengcheng) [reply pulsar-function-go/conf/conf.go]: added

**Review by cbornet — APPROVED** [review]

**Pull request merged**